src/nvim/api/private/helpers.c

```diff
@@ -885,6 +903,24 @@ static void init_type_metadata(Dictionary *metadata)
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
 
+String copy_string(String str)
+{
+  if (str.data != NULL) {
+    return (String){ .data = xmemdupz(str.data, str.size), .size = str.size };
+  } else {
+    return (String)STRING_INIT;
+  }
+}
+
+Array copy_array(Array array)
```

Comment:
If one copies an array and not an object? like the generated code in bridge will do.